### DISCUSSION TITLE
Unstable SCANCODE

### DISCUSSION THREAD

**Discussion by cttpla** [spotlessmind1975/ugbasic#362] [Optimizations]
This code is expected to print always 1 on screen as long as the user holds any key, but the result is constantly swapping between 0 and 1, hence I suspect the check isn't running on each frame. 
ALSO: if i omit the LOOP statement, the compiler isn't throwing any error.
Thank you for veryfing and fixing this.

Please check this doc for the MO keyboard codes

http://collection.thomson.free.fr/code/articles/prehisto_bulletin/page.php?XI=0&XJ=7&fbclid=IwAR2P1RgZGr89p9oBUxRYdgbS1Ym9a5hXBNct_Bx9HOZVU_Tabdh5b5eDicI 

CLS
DO
k = SCANCODE
IF (k == 0) THEN
LOCATE 0,0:PRINT "0"
ELSE 
LOCATE 0,0:PRINT "1"
ENDIF
LOOP

**Comment by spotlessmind1975** [comment]
**Hello @cttpla and thank you for your message!**

This is a bug, and I opened issue #363
